features/steps/python-to-notebook.py

```diff
@@ -8,78 +8,100 @@
 
 from synbconvert import SynapseNotebookConverter
 
+nth_dict = {
+    "first": 1,
+    "second": 2,
+    "third": 3,
+    "4th": 4
+}
+
 
 @given("we have a Python file `{filename}` with the following statements")
-@given("we have a simply Python file `{filename}` with the following statements")
+@given("we have a simple Python file `{filename}` with the following statements")
 def step_impl(context, filename) -> None:  # noqa: F811
+
+    directory = os.path.dirname(f"{context.working_directory}/{filename}")
+    if not os.path.exists(directory):
+        os.makedirs(directory)
     with open(f"{context.working_directory}/{filename}", "w") as f:
         f.write(context.text)
 
     # store mentioned file context for later use
     context.files.append(filename)
 
 
-@when(
-    "we transform this file with `nbsynconvert to-notebook {source_file} {target_file}`."
-)
+@when("we transform this file with `synbconvert convert {source_file} {target_file}`.")
+@when("we transform the notebook with `synbconvert convert {source_file} {target_file}`.")
 def step_impl(context, source_file, target_file) -> None:  # noqa: F811
     source_file = f"{context.working_directory}/{source_file}"
     target_file = f"{context.working_directory}/{target_file}"
 
     converter = SynapseNotebookConverter()
-    converter.convert_python_file_to_synapse_notebook(source_file, target_file)
 
+    if source_file.endswith(".py") and target_file.endswith(".json"):
```

Comment:
don't we have this if-clause also somewhere in our actual code?

Reply:
We have this if-clause in `synbconvert/cli.py`. Would it be good style to use methods of our actual code in our tests?

Reply:
yes - isn't a test about testing actual code?  ;)

Reply:
sure, but from a bdd pov, wouldn't using an untested method may falsify the feature test?

Reply:
done